pandas/io/formats/excel.py

```diff
@@ -472,12 +493,14 @@ def __init__(
         self.na_rep = na_rep
         if not isinstance(df, DataFrame):
             self.styler = df
+            self.styler._compute()  # calculate applied styles
```

Comment:
why does this need to be called? e.g. shouldn't this be in render?

Reply:
https://github.com/pandas-dev/pandas/pull/41995#discussion_r705286971